src/backend/distributed/transaction/transaction_management.c

```diff
@@ -328,12 +328,15 @@ CoordinatedTransactionCallback(XactEvent event, void *arg)
 				CoordinatedRemoteTransactionsAbort();
 			}
 
-			/* close connections etc. */
-			if (CurrentCoordinatedTransactionState != COORD_TRANS_NONE)
-			{
-				ResetPlacementConnectionManagement();
-				AfterXactConnectionHandling(false);
-			}
+			/*
+			 * Close connections etc. Contrary to a successful transaction we reset the placement connection management
+			 * irregardless of state of the statemachine as recorded in CurrentCoordinatedTransactionState.
+			 * The hashmaps recording the connection management live a memory context higher compared to most of the
+			 * data referenced in the hashmap. This causes use after free errors when the contents are retained due to
+			 * an error caused before the CurrentCoordinatedTransactionState changed.
+			 */
```

Comment:
very minor, I really like to not exceed 78 characters:

```suggestion
			/*
			 * Close connections etc. Contrary to a successful transaction,
			 * we reset the placement connection management regardless
			 * of state of the connection state-machine recorded in
			 * CurrentCoordinatedTransactionState.
			 * The hash-maps recording the connection management live in
			 * a memory context higher compared to most of the data referenced
			 * in the hash-map. This causes use after free errors when the contents
			 * are retained due to an error caused before
			 * CurrentCoordinatedTransactionState changed.
			 */
```

Reply:
don't know where 78 comes from.
Our, not often referenced, code style says:
```
We wrap to a new line after 90 characters.
```

https://docs.google.com/document/d/104XKGvh8Z_AamjtUEmbzNKgx0F6qsKyqIOJD_QRDJzI/edit#

currently these lines reach 120 charactes. I will wrap them according to our specification.
There are already lines just below reaching 80+ characters, so 78 is definitely not used here.